app/user.go

```diff
@@ -596,6 +596,26 @@ func (a *App) GetUsersWithoutTeam(offset int, limit int) ([]*model.User, *model.
 	return result.Data.([]*model.User), nil
 }
 
+// GetUsersPermittedToTeam returns the users who are permitted to join a team based on group constraints. If the
+// given team is not group-constrained then an empty slice will be returned.
```

Comment:
As a developer using the API, how can I differentiate between a not group constrained team / channel and a one that is Group constrained, but the group is empty? (If that is a realistic scenario)

Reply:
It's a good point that I was debating but obviously haven't satisfied yet. My thinking had been that it's the responsibility of the developer to verify that the team is group constrained first. However, that approach should be consistent with the name and the response. Perhaps it should return an error reflecting that the parameter was a non-group-constrained `teamID`. And perhaps it should be renamed accordingly, for example, `GetUsersPermittedToGroupConstrainedTeam`.